tests/draft2020-12/propertyNames.json

```diff
@@ -81,5 +84,60 @@
                 "valid": true
             }
         ]
+    },
+    {
+        "description": "propertyNames with additionalProperties",
```

Comment:
These test cases aren't actually testing whether additional properties are allowed.

Reply:
This test is to check if `additionalProperties` affect `propertyNames`.. so can you recheck on that

Reply:
"In this test case, the validation of data fails. For example:
```json  
{
            "description": "property name is invalid",
            "data": { "watermelon": 6 },
            "valid": false
}
```
  This failure occurs due to the exceeded property length in the validation of propertyName, rather than due to unevaluated properties. Therefore, this test does not actually determine whether additional properties are allowed or not."

Reply:
So, "`propertyNames `doesn't interact with other keywords." addressed in https://github.com/json-schema-org/JSON-Schema-Test-Suite/issues/305 this issue should be included under additional/unevaluated Properties??

Reply:
> This test is to actually check if additionalProperties affect propertyNames

No, that's backward. There's no reason `additionalProperties` would influence `propertyNames`, but the spec _could_ be misread that `propertyNames` defines properties that `additionalProperties` would then skip.  That's what the issue is about.

For example, your first test case with "apple" should be invalid because the property isn't allowed.

Reply:
So, we should be checking whether `propertyNames `affect `adiditionalProperties`. Got it, I feel we can go both the way on the tests if its ok.

Reply:
> I feel we can go both the way on the tests if its ok.

There's no need. There's nothing in the spec that suggests `additionalProperties` could affect `propertyNames`.  We don't have tests that ensure `maxLength` and `items` are independent _because_ nothing implies that they're connected.

Reply:
I see. I get it! so tests are written where the information in ambiguous, especially in spec..

Reply:
Yes, that and explicit requirements.

Reply:
Cool cool! Can you review the latest changes...